processor/aggregationprocessor/config.go

```diff
@@ -0,0 +1,114 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package aggregationprocessor provides a processor that samples pdata base level objects.
+package aggregationprocessor
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/observiq/observiq-otel-collector/processor/aggregationprocessor/internal/aggregate"
+	"go.uber.org/multierr"
+)
+
+// Config is the configuration for the processor
+type Config struct {
+	Interval time.Duration `mapstructure:"interval"`
+	// Include is a regex that must match the metric name for it to be sampled.
+	// Otherwise, the metric is passed through.
+	Include string `mapstructure:"include"`
+	// List of aggregations for the metric
+	Aggregations []AggregateConfig `mapstructure:"aggregations"`
+}
+
+// AggregateConfig is a config that
+type AggregateConfig struct {
+	// Type of aggregation
+	Type aggregate.AggregationType `mapstructure:"type"`
+	// MetricName is the name for the re-emitted metric. Defaults to `$0` (this is what is matched by the regex)
+	MetricName string `mapstructure:"metric_name"`
+}
+
+// Validate validate the config, returning an error explaining why it isn't if the config is invalid.
+func (a AggregateConfig) Validate() error {
+	var errs error
+
+	if !a.Type.Valid() {
+		errs = multierr.Append(errs, fmt.Errorf("invalid aggregate type for `type`: %s", a.Type))
+	}
+
+	return errs
+}
+
+// MetricNameString returns the configured name for the emitted metric, or "$0" if none was specified.
+func (a AggregateConfig) MetricNameString() string {
+	if a.MetricName != "" {
+		return a.MetricName
+	}
+
+	return "$0"
+}
+
+// Validate validates the processor configuration
+func (cfg Config) Validate() error {
+	var errs error
```

Comment:
I think in the past we've preferred to just exit on the first failure instead of building them up with multierr. This does differ from how OTel does it. I'm not to picky on this but would like to try to be consistent within our codebase.

Reply:
I've removed the multierr here. I do think the multierr stuff is really valuable, the less times you have to reconfigure the collector the better, but it's unrelated to this PR.